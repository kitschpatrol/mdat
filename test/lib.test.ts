import { toc } from '../dist'
import fs from 'node:fs/promises'
import { expect, it } from 'vitest'

it('should generate a table of contents', async () => {
	const doc = await fs.readFile('./test/assets/readme-basic.md', 'utf8')
	const value = await toc(doc)

	expect(value).toMatchSnapshot()
})
