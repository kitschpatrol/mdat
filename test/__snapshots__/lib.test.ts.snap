// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should generate a table of contents 1`] = `
"<!-- header -->

# Pluto

Pluto is a dwarf planet in the Kuiper belt.

## Contents

* [History](#history)
  * [Discovery](#discovery)
  * [Name and symbol](#name-and-symbol)
  * [Planet X disproved](#planet-x-disproved)
* [Orbit](#orbit)

## History

### Discovery

In the 1840s, Urbain Le Verrier used Newtonian mechanics to predict the position of…

### Name and symbol

The name Pluto is for the Roman god of the underworld, from a Greek epithet for Hades…

### Planet X disproved

Once Pluto was found, its faintness and lack of a viewable disc cast doubt…

## Orbit

Pluto's orbital period is about 248 years…
"
`;
